src/tds/bulk.c

```diff
@@ -258,6 +271,23 @@ tds7_build_bulk_insert_stmt(TDSSOCKET * tds, TDSPBCB * clause, TDSCOLUMN * bcpco
 	return TDS_SUCCESS;
 }
 
+static const char * uc_str(const char * s)
+{
+	/* return strupper(strdup(s)); */
+	size_t n = strcspn(s, "abcdefghijklmnopqrstuvwxyz");
+	if (s[n] == '\0') {
+		return s; /* already all uppercase */
+	} else {
+		char * result = malloc(n + 1);
```

Comment:
Are you sure about this ? Isn't this function converting ASCII to upper case but stopping at first lower case character ?

Reply:
Oops, yeah, I need to reset `n` to the length before proceeding here.

Reply:
Alternatively, the calling code could use `strcasestr`, but that's not portable, so we'd need to add an implementation to `replacements`.